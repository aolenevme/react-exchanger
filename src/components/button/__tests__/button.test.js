import React from "react";
import renderer from "react-test-renderer";

import Button from "../button.js";

describe("<Button />", () => {
    it("is rendered correctly", () => {
        expect(renderer
            .create(<Button isDisabled/>)
            .toJSON()).toMatchSnapshot();

        expect(renderer
            .create(<Button isDisabled={false}/>)
            .toJSON()).toMatchSnapshot();
    });

    describe("events", () => {
        it("reacts to onClick", () => {
            const onClick = jest.fn();

            const tree = renderer
                .create(<Button onClick={onClick}/>)
                .toJSON();

            tree.props.onClick();

            expect(onClick).toHaveBeenCalled();
        });
    });
});
