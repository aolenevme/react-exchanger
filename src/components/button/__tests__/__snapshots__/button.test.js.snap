// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> is rendered correctly 1`] = `
.c0 {
  border-color: transparent;
  background-color: transparent;
  color: #5591f7;
  -webkit-tap-highlight-color: transparent;
  font-size: 1rem;
  cursor: pointer;
  -webkit-transition: color 0.1s ease 0s;
  transition: color 0.1s ease 0s;
}

.c0:active {
  color: #5591f7;
}

.c0:disabled {
  cursor: unset;
}

.c0:hover {
  color: #5591f7;
}

<button
  className="c0"
  disabled={true}
  onClick={[Function]}
  type="button"
/>
`;

exports[`<Button /> is rendered correctly 2`] = `
.c0 {
  border-color: transparent;
  background-color: transparent;
  color: #f3f6ff;
  -webkit-tap-highlight-color: transparent;
  font-size: 1rem;
  cursor: pointer;
  -webkit-transition: color 0.1s ease 0s;
  transition: color 0.1s ease 0s;
}

.c0:active {
  color: #5591f7;
}

.c0:disabled {
  cursor: unset;
}

.c0:hover {
  color: #5591f7;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;
